packages/app/src/specs/SpecItem.vue

```diff
@@ -28,11 +28,16 @@
 </template>
 
 <script lang="ts" setup>
+import { toRefs } from '@vueuse/core'
 import HighlightedText from './HighlightedText.vue'
+import { useSplitIndexes } from './spec-utils'
 
-withDefaults(defineProps<{
+const props = withDefaults(defineProps<{
   fileName: string
   extension: string
   indexes?: number[]
 }>(), { indexes: () => [] })
+
+const refs = toRefs(props)
+const split = useSplitIndexes(refs.fileName, refs.indexes)
```

Comment:
It seems unusual for a `useXXX` function to return a `computed` property. You needed to do `props` -> `toRefs` -> `computed`. Also, it leads to the logic (when to split/join) mixing with the UI concerns (the reactivity layer).

Instead, why not just:

```js
const split = computed(() => {
  return props.indexes.reduce((acc, curr) => ...)
})
```

Since you want to share the logic and re-use it, just make a stateless function (no need to leak reactivity into it):

```js
const split = computed(() => {
  return deriveIndexes(props.fileName, props.indexes)
})
```

What do you think? Extra bonus of this is, if you decided you did want to unit test the `deriveIndexes` (eg, it became more complex) you wouldn't need a UI framework or reactivity system to do so.

Reply:
Your suggestion is exactly what I was looking for! Going to implement

Reply:
Refactored in [cb98862](https://github.com/cypress-io/cypress/pull/23325/commits/cb9886239b6523f78bdd8f29e5d8f03b473bf5c1)